### ISSUE TITLE
helm transitive dependencies affects values file key validation

### ISSUE THREAD

**Issue by m1o1** [helm/helm#8289]
I made a helm chart with prometheus-operator as a requirement (in requirements.yaml):
```
dependencies:
  - name: prometheus-operator
    version: "8.14.*"
    repository: https://kubernetes-charts.storage.googleapis.com/
```
This chart "wraps" this prometheus-operator helm chart to add some extra stuff. I want to specify some values with a "grafana" key (specific to my chart - no intention of propagating to dependencies), like so:
```
grafana:
  ingress:
    enabled: true
    # ... other values here
    tls:
      enabled: false
      secretName: ""
```

This fails validation with the following error message:
> Warning: Merging destination map for chart 'grafana'. The destination item 'tls' is a table and
ignoring the source 'tls' as it has a non-table value of: []

If I change the key to something other than `tls`, or if I change the top-level key to something other than `grafana`, it works. So I assume what's happening is it's thinking I want to set values on the grafana helm chart (which is a dependency of the prometheus-operator helm chart).

I don't have grafana in my requirements / dependencies, and if I try to set something like `grafana.adminPassword` directly from my `values.yaml` file, it does not actually get set, so I assume this is a bug and it's performing validation on keys for transitive dependencies. I would like to use the `grafana` key for my "wrapper" helm chart and have the ability to use keys that might exist in transitive subcharts without issue.

Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.7", GitCommit:"6c143d35bb11d74970e7bc0b6c45b6bfdffc0bd4", GitTreeState:"clean", BuildDate:"2019-12-11T12:42:56Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.7", GitCommit:"6c143d35bb11d74970e7bc0b6c45b6bfdffc0bd4", GitTreeState:"clean", BuildDate:"2019-12-11T12:34:17Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Minikube

Edit: To add, if it's desired that we should be able to set transitive subchart values with top-level keys in our values.yaml file (and perhaps there was some other reason why my adminPassword example didn't work), can I request a feature to disable that behavior (perhaps as a field in Chart.yaml) and require fully-qualified paths? I.e. to set grafana helm chart values, I would have to go through prometheus-operator, like `.Values.prometheus-operator.grafana.<subchartKey>`

**Comment by mattfarina** [comment]
In your edit you noted the issue. grafana is a dependency of the prometheus operator. You can see it at https://github.com/helm/charts/blob/master/stable/prometheus-operator/requirements.yaml.

In Helm v3 we cannot change this behavior as it would not work with keeping the same major version.

A short term solution could be to deploy grafana separately (a loose coupling of dependencies) and configure it to work with prometheus. Is there a reason for the tight coupling?

**Comment by Mister101** [comment]
I believe the prometheus operator used to have things separate (the operator / grafana / alertmanager etc) but they combined them into a single helm chart. I assume because of how much things are still churning it was easier to pin the different versions together and release as a single version. We actually did deploy grafana separately, but deploying the whole stack together seems easier to manage.

This seemed like a bug to me, since I can't actually set values on the sub-subchart (using the top-level key) - it just performs the validation and rejects if it fails. When I set values though, they are ignored. I've tested it with other grafana values now as well. If this isn't reproducible, I can try to make a smaller example to do so.
